src/java/org/apache/cassandra/db/lifecycle/LogFile.java

```diff
@@ -66,7 +66,7 @@
     private final LogReplicaSet replicas = new LogReplicaSet();
 
     // The transaction records, this set must be ORDER PRESERVING
-    private final LinkedHashSet<LogRecord> records = new LinkedHashSet<>();
+    private final Set<LogRecord> records = Collections.synchronizedSet(new LinkedHashSet<>()); // TODO: Hack until we fix CASSANDRA-14554
```

Comment:
I'm not very well familiar with `LifeCycleTransaction` code, but I'm not sure this is sufficient as a workaround. @belliottsmith Could you please confirm that this - or isn't - sufficient? Here and the equivalent change to `LogReplicaSet`.

Reply:
@dineshjoshi Ah. I see, that makes sense. The comment made it seem like this was supposed to be the fix for the issue, but this is just for the tests to pass? I'm good with that, given the extra context.

Reply:
I should've worded the comment differently but it is a hack to avoid a `ConcurrentModificationException` ([CASSANDRA-14554](https://issues.apache.org/jira/browse/CASSANDRA-14554)). This currently exists in trunk and was not introduced as part of this PR hence the separate ticket.